syntax/go.vim

```diff
@@ -366,13 +366,27 @@ if g:go_highlight_build_constraints != 0
   hi def link goPackageComment    Comment
 endif
 
-" :GoSameIds
-hi def goSameId term=bold cterm=bold ctermbg=white ctermfg=black
-
 " :GoCoverage commands
 hi def link goCoverageNormalText Comment
-hi def      goCoverageCovered    ctermfg=green guifg=#A6E22E
-hi def      goCoverageUncover    ctermfg=red guifg=#F92672
+
+function! s:hi()
+  " :GoSameIds
+  if &background == 'dark'
+    hi def goSameId term=bold cterm=bold ctermbg=white ctermfg=black guibg=white guifg=black
+  else
+    hi def goSameId term=bold cterm=bold ctermbg=14 guibg=Cyan
+  endif
+
+  " :GoCoverage commands
+  hi def      goCoverageCovered    ctermfg=green guifg=#A6E22E
+  hi def      goCoverageUncover    ctermfg=red guifg=#F92672
+endfunction
+
+augroup vim-go-hi
+  autocmd!
+  autocmd ColorScheme * call s:hi()
```

Comment:
There are a few different scenarios:

- Perhaps the most common case is when you reload the vimrc file with `:source ~/.vim/vimrc`. This will run the `colorscheme` command but won't re-run the syntax files (you can use `:e` to fix this though, but it's annoying).

- I've seen some people use different colour schemes for different filetypes − this autocommand will be run after the syntax file and there is no real way to fix this other than than the above `ColorScheme` autocommand.

- As you mentioned, it helps people who change colour schemes while Vim is running. I know some people who like to switch between a dark and light version depending on ambient light conditions. Again, `:e` will fix this, but it's annoying.